scipy/spatial/transform/rotation.py

```diff
@@ -1463,6 +1464,73 @@ def magnitude(self):
         else:
             return angles
 
+    def reduce(self, left=None, right=None, return_best=False):
+        """Reduce this rotation with the provided rotation groups.
+
+        Parameters
+        ----------
+        left : `Rotation` instance
+            Object containing the left rotation group.
+        right : `Rotation` instance
+            Object containing the right rotation group.
+        right : bool
+            Whether to return the best indices
+
+        Returns
+        -------
+        reduced : `Rotation` instance
+            Object containing reduced rotations.
+        best_indices: integer array
+            Best indices of `left` and `right`
+        """
+        if left is None and right is None:
+            return cls(self._quat, normalized=True, copy=True)
+        elif right is None:
+            right = Rotation.from_quat([[0, 0, 0, 1.]])
+        elif left is None:
+            left = Rotation.from_quat([[0, 0, 0, 1.]])
+
+        p = self
```

Comment:
Well, I didn't mean to substitute my scratch code as it is.

Reply:
We need to handle a situation when `left, self or right` contain a single rotation.

I believe adopted logic must be used here as well. If `self` is a single rotation, then returned reduced rotation also must be single. For `left` and `right` we can do broadcasting.

Reply:
I agree with your logic proposal.  I have added a test for reduction of single rotations.